checker/src/main/java/org/checkerframework/checker/index/upperbound/UpperBoundVisitor.java

```diff
@@ -269,6 +338,27 @@ private boolean relaxedCommonAssignmentCheck(
             return true;
         }
 
+        // Take advantage of information available on a HasSubsequence(a, from, to) annotation on either:
+        // 1. the lhs qualifier (varLtlQual): this allows us to show that iff varLtlQual includes LTL(b), b has HSS, and expQual includes LTL(a, -from), then the LTL(b) can be removed from varLtlQual
+        // 2. the rhs qualifier (expQual): this allows us to show that iff expQual includes LTL(b, from) for some b that has HSS, and varLtlQual includes LTL(a), then the LTL(a) can be removed from varLtlQual
```

Comment:
This rule seems wrong:
```
@HasSub(a, form, to) String[] b = new String[10];
int from = 2;
int to = 4;
@LTL(b, form) int rhs = 6;
@LTL(a) int lhs = rhs; // this assignment shouldn't be legal, but according to this rule it is.
```

Reply:
It is wrong. There's supposed to be one more check: that `rhs` is `@LessThan("to")`. I'm working on a fix.